c7n/resources/s3.py

```diff
@@ -724,12 +760,26 @@ def process_bucket(self, b):
         if p is None:
             return None
         p = json.loads(p)
+
         required = list(self.data.get('statement_ids', []))
         statements = p.get('Statement', [])
         for s in list(statements):
             if s.get('Sid') in required:
                 required.remove(s['Sid'])
-        if not required:
+
+        required_statements = list(self.data.get('statements', []))
+        for required_statement in required_statements:
+            for statement in statements:
+                found = False
+                for key, value in required_statement.items():
+                    if key in statement and value == statement[key]:
+                        found = True
```

Comment:
I think the change here is incorrect.
Provided the example policy in the comments above, the logic here says that any of 
key/value pairs in 
```
- Effect: Allow
  Action: 's3:*'
  Principal: '*'
```
is sufficient to match the statement rather than *all* key/value pairs matching.